### PR TITLE
fit function args test for nano keras multiprocessing training 

### PR DESCRIPTION
**Pull request by az2181036**

### CODE REVIEW

**Comment by az2181036** [comment]
Test results:
|           API           | results |                             link                             |
| :---------------------: | :-----: | :----------------------------------------------------------: |
| epochs, validation_data |  pass   | https://github.com/intel-analytics/BigDL/blob/2f727c72f81d5a1d219a02fd7f034a4f110e50a4/python/nano/test/tf/keras/test_fit_different_parameter.py#L85 |
|       batch_size        |  pass   | https://github.com/intel-analytics/BigDL/blob/2f727c72f81d5a1d219a02fd7f034a4f110e50a4/python/nano/test/tf/keras/test_fit_different_parameter.py#L93 |
|         verbose         |  pass   | https://github.com/intel-analytics/BigDL/blob/2f727c72f81d5a1d219a02fd7f034a4f110e50a4/python/nano/test/tf/keras/test_fit_different_parameter.py#L104 |
|        callbacks        |  fail   | https://github.com/intel-analytics/BigDL/blob/2f727c72f81d5a1d219a02fd7f034a4f110e50a4/python/nano/test/tf/keras/test_fit_different_parameter.py#L123 |
|    validation_split     |  fail   | https://github.com/intel-analytics/BigDL/blob/2f727c72f81d5a1d219a02fd7f034a4f110e50a4/python/nano/test/tf/keras/test_fit_different_parameter.py#L135 |
|         shuffle         |  pass   | https://github.com/intel-analytics/BigDL/blob/2f727c72f81d5a1d219a02fd7f034a4f110e50a4/python/nano/test/tf/keras/test_fit_different_parameter.py#L142 |
|      class_weight       |  fail   | https://github.com/intel-analytics/BigDL/blob/2f727c72f81d5a1d219a02fd7f034a4f110e50a4/python/nano/test/tf/keras/test_fit_different_parameter.py#L148 |
|      sample_weight      |  fail   | https://github.com/intel-analytics/BigDL/blob/2f727c72f81d5a1d219a02fd7f034a4f110e50a4/python/nano/test/tf/keras/test_fit_different_parameter.py#L161 |
|      initial_epoch      |  pass   | https://github.com/intel-analytics/BigDL/blob/2f727c72f81d5a1d219a02fd7f034a4f110e50a4/python/nano/test/tf/keras/test_fit_different_parameter.py#L172 |
|     steps_per_epoch     |  pass   | https://github.com/intel-analytics/BigDL/blob/2f727c72f81d5a1d219a02fd7f034a4f110e50a4/python/nano/test/tf/keras/test_fit_different_parameter.py#L179 |
|    validation_steps     |  pass   | https://github.com/intel-analytics/BigDL/blob/2f727c72f81d5a1d219a02fd7f034a4f110e50a4/python/nano/test/tf/keras/test_fit_different_parameter.py#L186 |
|  validation_batch_size  |  pass   | https://github.com/intel-analytics/BigDL/blob/2f727c72f81d5a1d219a02fd7f034a4f110e50a4/python/nano/test/tf/keras/test_fit_different_parameter.py#L93 |
|     validation_freq     |  pass   | https://github.com/intel-analytics/BigDL/blob/2f727c72f81d5a1d219a02fd7f034a4f110e50a4/python/nano/test/tf/keras/test_fit_different_parameter.py#L193 |
|     max_queue_size      |  pass   | https://github.com/intel-analytics/BigDL/blob/2f727c72f81d5a1d219a02fd7f034a4f110e50a4/python/nano/test/tf/keras/test_fit_different_parameter.py#L200 |
|         workers         |  pass   | https://github.com/intel-analytics/BigDL/blob/2f727c72f81d5a1d219a02fd7f034a4f110e50a4/python/nano/test/tf/keras/test_fit_different_parameter.py#L207 |
|   use_multiprocessing   |  pass   | https://github.com/intel-analytics/BigDL/blob/2f727c72f81d5a1d219a02fd7f034a4f110e50a4/python/nano/test/tf/keras/test_fit_different_parameter.py#L214 |

**Comment by az2181036** [comment]
callbacks: It will be fixed while changing 'pickle' to 'cloudpickle'.
validation_split: "validation_split argument is not supported with distribution strategies." [tensorflow github link](https://github.com/tensorflow/tensorflow/blob/c2782608abab13d3243b521cad45219540156d9a/tensorflow/python/keras/engine/training_distributed_v1.py#L645)
class_weight and sample_weight: we just support `tf.data.Dataset` for now. [link](https://github.com/intel-analytics/BigDL/blob/848662ca4b5736b0028f3f03f5f6fd9987ef5499/python/nano/src/bigdl/nano/tf/keras/training_utils.py#L72)
